src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs

```diff
@@ -174,10 +174,22 @@ public override string ToString()
         #region private
 
         // NotificationSource implementation
+
+        /// <summary>
+        /// Optional: If there is an expensive setup for the notification,
+        /// you may call IsEnabled() as the first and most efficient check before doing this setup. 
+        /// Producers may use this check before IsEnabled(string) in the most performance-critical parts of the system
+        /// to ensure somebody listens to the DiagnosticListener at all.
+        /// </summary>
```

Comment:
Nit: What's currently in the summary seems more like remarks to me.  For a summary, I'd expect something like "Determines whether there are any registered subscribers.", and then the remarks can go on to detail how the other IsEnabled overloads allow for more fine-grained filtering.